components/statusLabel/StatusLabel.js

```diff
@@ -0,0 +1,73 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import omit from 'lodash.omit';
+import theme from './theme.css';
+
+class StatusLabel extends Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    className: PropTypes.string,
+    mint: PropTypes.bool,
+    teal: PropTypes.bool,
+    violet: PropTypes.bool,
+    ruby: PropTypes.bool,
+    gold: PropTypes.bool,
+    aqua: PropTypes.bool,
+    small: PropTypes.bool,
+  };
+
+  getColor () {
+    const colors = [
+      'mint',
+      'violet',
+      'ruby',
+      'gold',
+      'aqua',
+      'teal',
+    ];
+
+    const color = colors.filter(color => this.props.hasOwnProperty(color));
+    return color[0];
```

Comment:
As I already commented somewhere else, i don't like the filter because you can't return "early", it keeps evaluating the rest even if we already found a match.

Also, `hasOwnProperty` will return `true`, even if the prop is `false` 👎

Reply:
Fair point, we could use `array.find` which returns on the first true evaluation.

And I guess the conditional changes to `this.props.hasOwnProperty(color) && this.props.color`

We could also ditch the hasOwnProperty check if we set default propTypes to false. It just that I had to triple look at the if to finally grasp what was going on.

Reply:
Implementation would change to:

`return colors.find(color => this.props.hasOwnProperty(color) && this.props.color);`

Reply:
Yours won't be correct though (the `this.props.color`) and I think we can just drop the hasOwnProperty, because undefined evaluates to false.

`return colors.find(color => this.props[color]);`